pos_barcode_tare/static/src/js/pos_barcode_tare.js

```diff
@@ -0,0 +1,199 @@
+            },
+
+            show: function () {
+                var self = this;
+                this._super();
+                this.pos.barcode_reader.set_action_callback(
+                    'weight',
+                    _.bind(self.barcode_weight_action, self));
+            },
+        });
+});
+
+odoo.define('tare-screen-button.button', function (require) {
+    "use strict";
+    var core = require('web.core');
+    var screens = require('point_of_sale.screens');
+    var gui = require('point_of_sale.gui');
+
+    var TareScreenButton = screens.ActionButtonWidget.extend({
+        template: 'TareScreenButton',
+
+        button_click: function () {
+            var self = this;
+            this.gui.show_screen('tare');
+        },
+    });
+
+    screens.define_action_button({
+        'name': 'tareScreenButton',
+        'widget': TareScreenButton,
+    });
+});
+
+odoo.define('tare-screen.screen', function (require) {
+    "use strict";
+    var chrome = require('point_of_sale.chrome');
+    var core = require('web.core');
+    var devices = require('point_of_sale.devices');
+    var gui = require('point_of_sale.gui');
+    var models = require('point_of_sale.models');
+    var screens = require('point_of_sale.screens');
+    var QWeb = core.qweb;
+
+    var TareScreenWidget = screens.ScreenWidget.extend({
+        template: 'TareScreenWidget',
+        next_screen: 'products',
+        previous_screen: 'products',
+        default_tare_value_kg: 0.0,
+
+        show: function () {
+            this._super();
+            var self = this;
+            var queue = this.pos.proxy_queue;
+
+            queue.schedule(function () {
+                return self.pos.proxy.scale_read().then(function (weight) {
+                    self.set_weight(weight.weight);
+                });
+            }, {duration:150, repeat: true});
+
+            this.render_receipt();
+            this.lock_screen(true);
+        },
+        set_weight: function (weight) {
+            if (weight > 0) {
+                this.weight = weight;
+                this.render_receipt();
+                this.lock_screen(false);
+            }
+        },
+        get_weight: function () {
+            if (typeof this.weight === 'undefined') {
+                return this.default_tare_value_kg;
+            }
+            return this.weight;
+        },
+        ean13_checksum: function (s) {
+            var result = 0;
+            for (var counter = s.length-1; counter >=0; counter--) {
+                var counterCheckSum = counter % 2;
+                counterCheckSum *= 2;
+                counterCheckSum += 1;
+                result += parseInt(s.charAt(counter), 10) * counterCheckSum;
+            }
+            var checksum = 10;
+            checksum -= result % 10;
+            return checksum % 10;
+        },
+        barcode_data: function (weight) {
+            var padding_size = 5;
+            var default_weight_prefix_id = "21";
```

Comment:
I think that it should be done with a ``barcode.nomenclature``, and not hardcoded.
Note : can be done in a second time, but should be written in the "Roadmap / Know issue" section.

Reply:
I'll change that!

Reply:
Agreed! I'll try to fix that using the nomenclature in this version.

Reply:
This is solved.